src/ucs/async/eventfd.c

```diff
@@ -0,0 +1,93 @@
+/**
+* Copyright (c) NVIDIA CORPORATION & AFFILIATES, 2023-2024. ALL RIGHTS RESERVED.
+*
+* See file LICENSE for terms.
+*/
+
+#ifdef HAVE_CONFIG_H
+#  include "config.h"
+#endif
+
+#include "eventfd.h"
+
+#include <ucs/debug/log.h>
+#include <ucs/sys/sys.h>
+
+
+typedef enum {
+    UCS_ASYNC_EVENTFD_POLL   = 0,
+    UCS_ASYNC_EVENTFD_SIGNAL = 1
+} ucs_async_eventfd_io_t;
+
+ucs_status_t ucs_async_eventfd_common_io(int fd, ucs_async_eventfd_io_t mode)
+{
+    uint64_t dummy = 1;
+    int ret;
+
+    do {
+        if (mode == UCS_ASYNC_EVENTFD_SIGNAL) {
+            ret = write(fd, &dummy, sizeof(dummy));
+        } else {
+            ret = read(fd, &dummy, sizeof(dummy));
+        }
+        if (ret == sizeof(dummy)) {
+            return UCS_OK;
+        }
+
+        if (ret == -1) {
+            if (errno == EAGAIN) {
+                if (mode == UCS_ASYNC_EVENTFD_POLL) {
+                    return UCS_ERR_NO_PROGRESS;
+                } else {
+                    continue;
```

Comment:
can you please rephactor this whole loop?
e.g. this `continue` statement will work as a `break`, because `while` check for `ret == 0` will file